ramanujan/cmf.py

```diff
@@ -22,17 +22,26 @@ def My(self, x, y) -> Matrix:
     def subs(self, substitutions):
         return CMF(self.m_Mx.subs(substitutions), self.m_My.subs(substitutions))
 
-    def trajectory_matrix(self, trajectory) -> Matrix:
+    def trajectory_matrix(self, trajectory, start=None) -> Matrix:
         m = self.m_Mx.walk({x: 1, y: 0}, trajectory[0], {x: x, y: y})
         m *= self.m_My.walk({x: 0, y: 1}, trajectory[1], {x: x + trajectory[0], y: y})
+        if start is not None:
+            m = CMF.substitute_trajectory(m, trajectory, start)
         return simplify(m)
 
+    @staticmethod
+    def substitute_trajectory(m: Matrix, trajectory, start):
```

Comment:
What is the purpose of this? Do you mean you start from a different location of just substitute x,y with n?

Reply:
This function takes a matrix and substitutes the trajectory into a single variable. For example, if you went on the diagonal (x = y), it will substitute both x for n and y for n. The substitution is well-defined given a starting point and a trajectory.